### PR TITLE
Alternate documentation layout

### PR DESCRIPTION
**Pull request by evankanderson**
Alternate version of #201 

See https://docs.google.com/document/d/1UKyjA0nfNyvDIiF86YEeskV5_pZ_kP07AoDHrDUq8Xo/edit#heading=h.los3tdhhh2fb for the general template, though I took a few liberties as I was actually writing it up.

# Changes

Adds end-user documentation for the RabbitMQ Source.

- 📖  End user documentation
- 📖  Moved contributor documentation to `DEVELOPMENT.md`

/kind documentation

**Release Note**

```release-note
Added user-facing documentation for the RabbitMQ source
```

**Docs**

```docs
Will update the sources docs to point to this page once merged.
```

### CODE REVIEW

File: source/README.adoc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+== Reference
+
+=== Prerequisites
+
+* A https://www.rabbitmq.com/[RabbitMQ] installation. On Kubernetes, you can use
+https://www.rabbitmq.com/kubernetes/operator/operator-overview.html[the RabbitMQ
+operator] to set up a RabbitMQ installation.
+
+* An understanding of RabbitMQ concepts like Brokers, Exchanges, and Queues.
```
Comment (abrennan89) [inline source/README.adoc:96]: I'd move this to the start :)

Comment (abrennan89) [reply source/README.adoc:96]: Otherwise lgtm

**Comment by vaikas** [comment]
@evankanderson what's the plan with this PR? Looks like it needs a rebase at least.

**Comment by vaikas** [comment]
@evankanderson just a check on what you'd like to do with this one?

**Comment by evankanderson** [comment]
/close

We chose #201 instead

**Pull request closed without merging**